sdks/python/apache_beam/runners/interactive/display/pipeline_graph_test.py

```diff
@@ -83,6 +91,45 @@ def test_get_dot(self):
          '}\n'),
         pipeline_graph.PipelineGraph(p).get_dot())
 
+  @patch('IPython.get_ipython', mock_get_ipython)
+  def test_get_dot_within_notebook(self):
+    # Assume a mocked ipython kernel and notebook frontend have been set up.
+    ie.current_env()._is_in_ipython = True
+    ie.current_env()._is_in_notebook = True
+    with mock_get_ipython():  # Cell 1
+      p = beam.Pipeline(ir.InteractiveRunner())
+      # Immediately track this local pipeline so that ipython prompts when
+      # applying transforms will be tracked and used for labels.
+      ib.watch(locals())
+
+    with mock_get_ipython():  # Cell 2
+      init_pcoll = p | 'Init' >> beam.Create(range(10))
+
+    with mock_get_ipython():  # Cell 3
+      squares = init_pcoll | 'Square' >> beam.Map(lambda x: x * x)
+
+    with mock_get_ipython():  # Cell 4
+      cubes = init_pcoll | 'Cube' >> beam.Map(lambda x: x ** 3)
+
+    # Tracks all PCollections defined so far.
+    ib.watch(locals())
+    self.assertEqual(
+        ('digraph G {\n'
+         'node [color=blue, fontcolor=blue, shape=box];\n'
+         '"Cell 2: Init";\n'
+         'init_pcoll [shape=circle];\n'
+         '"Cell 3: Square";\n'
+         'squares [shape=circle];\n'
+         '"Cell 4: Cube";\n'
+         'cubes [shape=circle];\n'
+         '"Cell 2: Init" -> init_pcoll;\n'
+         'init_pcoll -> "Cell 3: Square";\n'
+         'init_pcoll -> "Cell 4: Cube";\n'
+         '"Cell 3: Square" -> squares;\n'
+         '"Cell 4: Cube" -> cubes;\n'
+         '}\n'),
```

Comment:
Should we add a test for composite transforms?

Reply:
`Create` is a composite transform. The dot based pipeline graph only renders top-level transform and our label altering only happens to top-level transforms. So whether a transform is composite or not doesn't affect neither.